controllers/openstackcluster_controller.go

```diff
@@ -178,6 +179,9 @@ func (r *OpenStackClusterReconciler) reconcileDelete(ctx context.Context, scope
 			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete load balancer: %w", err), false)
 			return reconcile.Result{}, fmt.Errorf("failed to delete load balancer: %w", err)
 		}
+		// Once a loadbalancer is deleted, we need to wait for the Octavia management ports to be cleaned up
+		// by Octavia before we can proceed with the deletion of the network components.
+		return reconcile.Result{RequeueAfter: waitForOctaviaPortsCleanup}, nil
```

Comment:
If it's more efficient we could poll here for a short period of time before:
* returning with an error?
* requeuing?

Reply:
I'm trying to tackle this problem from a different angle; I'll report back my results.

Reply:
i've pushed a separate commit, please re-review.